File: examples/openai-sdk/conversation.js
// Пример использования OpenAI SDK для диалога с несколькими сообщениями
// Установка: npm install openai

import OpenAI from 'openai';

const openai = new OpenAI({
    baseURL: 'http://localhost:3264/api', 
    apiKey: 'dummy-key', // Ключ не используется, но требуется для SDK
});

async function conversationExample() {
    try {
        console.log('Начинаем диалог с Qwen AI...\n');
        
        // Первое сообщение пользователя
        console.log('Пользователь: Привет! Расскажи о квантовой физике простыми словами.');
        
        let completion = await openai.chat.completions.create({
            messages: [
                { role: 'user', content: 'Привет! Расскажи о квантовой физике простыми словами.' }
            ],
            model: 'qwen-max-latest',
        });
        
        const assistantResponse1 = completion.choices[0].message.content;
        console.log('\nQwen:', assistantResponse1);
        
        // Второе сообщение пользователя, включающее историю беседы
        console.log('\nПользователь: А как это связано с теорией относительности?');
        
        completion = await openai.chat.completions.create({
            messages: [
                { role: 'user', content: 'Привет! Расскажи о квантовой физике простыми словами.' },
                { role: 'assistant', content: assistantResponse1 },
                { role: 'user', content: 'А как это связано с теорией относительности?' }
            ],
            model: 'qwen-max-latest',
        });
        
        const assistantResponse2 = completion.choices[0].message.content;
        console.log('\nQwen:', assistantResponse2);
        
        // Третье сообщение пользователя
        console.log('\nПользователь: Спасибо! Кто из ученых внес наибольший вклад в развитие этих теорий?');
        
        completion = await openai.chat.completions.create({
            messages: [
                { role: 'user', content: 'Привет! Расскажи о квантовой физике простыми словами.' },
                { role: 'assistant', content: assistantResponse1 },
                { role: 'user', content: 'А как это связано с теорией относительности?' },
                { role: 'assistant', content: assistantResponse2 },
                { role: 'user', content: 'Спасибо! Кто из ученых внес наибольший вклад в развитие этих теорий?' }
            ],
            model: 'qwen-max-latest',
        });
        
        console.log('\nQwen:', completion.choices[0].message.content);
        console.log('\nДиалог успешно завершен.');

    } catch (error) {
        console.error('Ошибка при выполнении диалога:', error);
    }
}

// Запуск
conversationExample(); 

File: examples/python-sdk/conversation.py
from openai import OpenAI

client = OpenAI(
    base_url="http://localhost:3264/api",
    api_key="free-qwen-api",
)

chat_id = None
parent_id = None

first = client.chat.completions.create(
    model="qwen-max-latest",
    messages=[{"role": "user", "content": "Меня зовут Алексей. Запомни это."}],
)
print("Assistant:", first.choices[0].message.content)

if hasattr(first, "chatId"):
    chat_id = first.chatId
if hasattr(first, "parentId"):
    parent_id = first.parentId

second_messages = [{"role": "user", "content": "Как меня зовут?"}]
extra_body = {}
if chat_id:
    extra_body["chatId"] = chat_id
if parent_id:
    extra_body["parentId"] = parent_id

second = client.chat.completions.create(
    model="qwen-max-latest",
    messages=second_messages,
    extra_body=extra_body or None,
)
print("Assistant:", second.choices[0].message.content)
